Add teamup-rooms package

drewwatkins11 opened the pull request:
## Description

This adds an extension to book rooms in Teamup. Teamup has many use cases for scheduling, but this specifically targets instances such as conference rooms or call rooms in a coworking space.

## Screencast

[https://www.loom.com/share/1f244786cb424b80ae0be99ca103c9e0](https://www.loom.com/share/1f244786cb424b80ae0be99ca103c9e0)

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

pernielsentikaer commented on extensions/teamup-rooms/package.json:
```diff
@@ -0,0 +1,65 @@
+    {
+      "name": "index",
+      "title": "Check for Open Rooms",
+      "description": "Check what rooms are open right now",
```
```suggestion
      "description": "Check what rooms are open right now and quick book them",
```

pernielsentikaer commented on extensions/teamup-rooms/src/index.tsx:
```diff
@@ -0,0 +1,254 @@
+
+  return (
+    <List
+      navigationTitle={`Rooms open in the next ${timeString}`}
```
```suggestion
```
Don't change the navigationTitle in the root command

drewwatkins11 replied:
@pernielsentikaer No problem, I'll make this change. Has there been any discussion about whether a dynamic badge or feedback area could be added in the title area? It seems like a logical place for feedback since there isn't a design guideline for adding a heading or descriptive test in the list.

pernielsentikaer replied:
@drewwatkins11 At the moment it's done in most of the cases with a <section title={}> but it's something that will be discuessed later on again since it's something that pops up from time to time. Forgot to mention where I [quoted from](https://raycastcommunity.slack.com/archives/C01AC2X0GMN/p1655440176400979?thread_ts=1655219827.794579&cid=C01AC2X0GMN) 😄

pernielsentikaer reviewed (changes requested):
Hi @drewwatkins11 👋 

Thanks for your contribution, it looks like it could be quite handy! 😄 

It would be a nice if you can handle the rest of the errors gracefully and return an empty array instead 😄 

![CleanShot 2022-06-16 at 14 14 16@2x](https://user-images.githubusercontent.com/38033107/174067507-eddfd65d-641a-4da6-8bc9-d2f726cbcce1.png)

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

pernielsentikaer reviewed (commented):
Hi @drewwatkins11 👋 

Thanks for the corrections, it looks perfect 💪 

Something that I miss is a README. There is no way to know where to get the API key. I found the place to retrieve the API key, but I'm struggling to find the correct "Calendar ID"  - we can debug this in [Slack](https://www.raycast.com/community) or another place or just share a screenshot with a fat arrow 😄

drewwatkins11 commented:
@pernielsentikaer No problem. I renamed the `readme.md` file to `README.md` and added a screenshot to make it more clear where to get the calendar URL. I also snuck in another functionality update to support password-protected calendars.

If you choose to test a password-protected calendar, you'll have to signup for a free trial of Teamup's paid plan (no billing information is needed). You can then add a password when creating the calendar share link ([https://calendar.teamup.com/kb/password-protection/](https://calendar.teamup.com/kb/password-protection/)).

pernielsentikaer reviewed (approved):
Hi @drewwatkins11 👋 

Thanks for you contribution; I think we're ready to merge it 🔥 

Thanks for your corporation; it was a pleasure to work with you 💪
